onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorConvolution.cpp

```diff
@@ -32,7 +35,8 @@ class DmlOperatorConvolution : public DmlOperator, public ConvolutionHelperBase
         m_inputTensorDescs[1] = CreateTensorDescFromInput(kernelInfo, 1, TensorAxis::DoNotCoerce, TensorAxis::NoPlacementAdjustment, NonspatialDimensionCount, std::nullopt);
 
         // Bias is optional so only adjust it if it exists.
-        if (kernelInfo.GetInputCount() > 2)
+        if ((!hasDynamicPads && kernelInfo.GetInputCount() > 2) ||
+            (hasDynamicPads && kernelInfo.GetInputCount() > 3))
```

Comment:
It looks like this (or similar) conditional is repeated a few times. Can it be simplified by doing it once, and assigning to a bool "hasBiasTensor" or similar? That would make it easier to read too.